[Yap Zhi Heng] iP

Polygonalr opened the pull request:
# 💩This is a heading💩

* this is bullet point 1
* this is bullet point 2

## This is a heading 2

1. this is numbered item 1
2. this is numbered item 2
3. this is numbered item 3

## This is another heading 2

- [ ] a task list item
- [x] a completed task list item

*Here's `Hello World` in [**Emojicode**](https://www.emojicode.org/).*

```emojicode
🏁 🍇
  😀 🔤Hello, World!🔤❗️💭 Prints "Hello, World!"
🍉
```

In case Emojicode does not fulfill the requirement, here's the equivalent in ~~C~~ **Python**.

```python3
print("Hello, World!")
```

>"If anyone here believes in telekinesis, raise my hand" ~Kurt Vonnegut

Commit:
Add Gradle support

Polygonalr committed:
Implement level 1

Polygonalr committed:
Implement level 2

Polygonalr committed:
Implement level 3

Polygonalr committed:
Implement level 4

Polygonalr committed:
Implement automated testing

Polygonalr committed:
Refactor code to use ArrayList to store Tasks

Polygonalr committed:
Refactor code to loop in main

Polygonalr committed:
Implement level 5

Polygonalr committed:
Implement level 6

Polygonalr committed:
Implement save-load function

Polygonalr committed:
Implement java.time.LocalDate for Deadlines and Events

Polygonalr committed:
Merge branch 'branch-Level-7'

Polygonalr committed:
Merge branch 'branch-Level-8'

Polygonalr committed:
Implement OOP and packages

Polygonalr committed:
Implement equals func for tasks and commands

Polygonalr committed:
Fix class name in gradle

Polygonalr committed:
Update tasks to use DukeException

Polygonalr committed:
Implement JUnit tests

Polygonalr committed:
Configure shadowJar

Polygonalr committed:
Add JavaDoc to main classes

Polygonalr committed:
Update code style to match standard

Polygonalr committed:
Implement find command

Polygonalr committed:
Merge tag 'A-JavaDoc'

Polygonalr committed:
Merge tag 'A-CodingStandard'

Polygonalr committed:
Merge branch 'branch-Level-9'

Polygonalr committed:
Add JavaDocs for find command

Polygonalr committed:
Refactor Command classes

Update Command objects to return a message after executing
Update Duke class to print the messages returned by Commands after executing

Polygonalr committed:
Implement initial GUI with JavaFX

Polygonalr committed:
Update gitignore to ignore vscode configs

Hoodineee commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,72 @@
+     */
+    public static Deadline fromSaveString(String saveString) throws DukeException {
+        String[] splitSaveString = saveString.split("(\",\")|(\",)|(,\")|\"");
+        if(splitSaveString.length != 3) {
```
It would be nice if there is a spacing between if and the opening bracket.

Polygonalr replied:
Thanks for pointing it out! I've fixed it (and the subsequent missing spaces) in my latest commit.

Hoodineee commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,72 @@
+     */
+    public static Event fromSaveString(String saveString) throws DukeException {
+        String[] splitSaveString = saveString.split("(\",\")|(\",)|(,\")|\"");
+        if(splitSaveString.length != 3) {
```
It would be good if you had a spacing between the if and open bracket here as well.

Hoodineee commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,74 @@
+package duke;
+import java.util.ArrayList;
```
It would be good if there was a line spacing between the package and the imports.

Polygonalr replied:
I've fixed this in the latest commit as well, thanks for letting me know!

Hoodineee commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,98 @@
+     */
+    public static Task fromSaveString(String saveString) throws DukeException {
+        String[] splitSaveString = saveString.split("(\",\")|(\",)|(,\")|\"");
+        if(splitSaveString.length != 2) {
```
It would be good if there was a spacing between the if and open bracket.

Hoodineee commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,30 @@
+     * Converts TaskList into save data string to be stored in a file.
+     * @return the save data string
+     */
+    public String toSaveData() {
+        return this.stream().map(Task::toSaveData).collect(joining("\n"));
+    }
+
+    /**
+     * Finds all tasks in the list that contain the given string.
+     * @param toFind the string to be searched within task descriptions
+     * @return list of tasks that contain the given string
+     */
+    public ArrayList<Task> find(String toFind) {
+        return this.stream().filter(task -> task.descriptionContains(toFind)).collect(ArrayList::new, ArrayList::add, ArrayList::addAll);
+    }
+}
```
Very good use of map and filter!

Hoodineee reviewed (commented):
Overall really good codes. Very easy to read and just minor things missed out like spaces. LGTM!

Polygonalr committed:
Implement and configure CheckStyle

Add CheckStyle plugin to gradle build file.
Lint existing code to fulfill checkstyle criteria.

Polygonalr committed:
Merge branch 'branch-Level-10'

Polygonalr committed:
Merge branch 'branch-A-CheckStyle'

Polygonalr committed:
Add some assertions

Polygonalr committed:
Abstract out DukeException class

There are many re-use of the same exception messages within the code.
This commit aims to reduce the repetition of exception messages by
defining new exception classes which inherits DukeException.

Polygonalr committed:
Abstract magic characters for tasks

Use of magic variables are to be avoided due to possible regression. This
commit aims to resolve a few instances of such usage, specifically within
the Task class and its child classes.

Polygonalr committed:
Fix checkstyle issues

Did not run checkstyleMain for previous few commits, so the apparent
checkstyle issues is resolved this commit.

Polygonalr committed:
Merge pull request #1 from Polygonalr/branch-A-Assertions

Add some assertions

Polygonalr committed:
Merge branch 'master' into branch-A-CodeQuality

Polygonalr committed:
Merge pull request #2 from Polygonalr/branch-A-CodeQuality

Improve code quality

Polygonalr committed:
Add GitHub CI

Also fixes another checkstyle oversight in ParserTest.

Polygonalr committed:
Fix some assert lines

Polygonalr committed:
Initial implementation of notes

`note` and `list` command supports it at the moment.
Support for other commands and saving/loading from disk to be added.

Polygonalr committed:
Add delete command support for notes

Polygonalr committed:
Update how list objects are loaded

This is to support loading of other list objects such as notes.

Polygonalr committed:
Implement saving/loading of notes from storage

Polygonalr committed:
Add find command support for note

Implement find command in parser as well.

Polygonalr committed:
Update JavaDocs to include support for Notes

Polygonalr committed:
Implement unit tests for Notes

Polygonalr committed:
Merge pull request #3 from Polygonalr/branch-BCD-Extension

Feature: Notes

Polygonalr committed:
Fix all code quality issues

Resolves #4

Polygonalr committed:
Improve GUI

Add unique avatar for user.
Add drop shadow for avatars.

Polygonalr committed:
Add emoji support

Polygonalr committed:
GUI: Enclose dialogs within cards

Polygonalr committed:
Add more personality to the chatbot

Now it is called Caramel, that sends friendlier messages.

Polygonalr committed:
Add user guide and documentation

Polygonalr committed:
Add indication for whether save file is loaded

Polygonalr committed:
Fix header comments verb forms
